taipy/core/data/aws_s3.py

```diff
@@ -109,23 +99,15 @@ def __init__(
         super().__init__(
             config_id,
             scope,
-            id,
-            owner_id,
-            parent_ids,
-            last_edit_date,
-            edits,
-            version or _VersionManagerFactory._build_manager()._get_latest_version(),
-            validity_period,
-            edit_in_progress,
-            editor_id,
-            editor_expiration_date,
             **properties,
         )
 
         self._s3_client = boto3.client(
             "s3",
             aws_access_key_id=properties.get(self.__AWS_ACCESS_KEY_ID),
             aws_secret_access_key=properties.get(self.__AWS_SECRET_ACCESS_KEY),
+            region_name=properties.get(self.__AWS_REGION), 
+            **properties.get(self.__AWS_S3_OBJECT_PARAMETERS, {}),
```

Comment:
The arguments for [`boto3.client()`](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/core/session.html#boto3.session.Session.client) and [`S3.Client.get_object()`](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/s3/client/get_object.html) and [`S3.Client.put_object()`](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/s3/client/put_object.html) are all different.

Here, using the same parameters for all methods will not work.

We can:
- have a different set of parameters for each method, like `__AWS_S3_CLIENT_PARAMETERS`, `__AWS_S3_GET_OBJECT_PARAMETERS`, and `__AWS_S3_PUT_OBJECT_PARAMETERS`
- take a look into the list of parameters for each method, decide which parameters are the most important, and only expose those.

I would suggest the former approach, since it's the most generic.